api/base/serializers.py

```diff
@@ -143,92 +171,142 @@ def get_attribute(self, obj):
         return field_source_method(auth=auth)
 
 
-class JSONAPIHyperlinkedIdentityField(ser.HyperlinkedIdentityField):
+class RelationshipField(ser.HyperlinkedIdentityField):
     """
-    HyperlinkedIdentityField that returns a nested dict with url,
-    optional meta information, and link_type.
+    RelationshipField that permits the return of both self and related links, along with optional
+    meta information.
 
     Example:
+    children = RelationshipField(
+        related_view='nodes:node-children',
+        related_view_kwargs={'node_id': '<pk>'},
+        self_view='nodes:node-node-children-relationship',
+        self_view_kwargs={'node_id': '<pk>'},
+        related_meta={'count': 'get_node_count'}
+    )
 
-        children = JSONAPIHyperlinkedIdentityField(view_name='nodes:node-children', lookup_field='pk',
-                                    link_type='related', lookup_url_kwarg='node_id', meta={'count': 'get_node_count'})
+    The lookup field must be surrounded in angular brackets to find the attribute on the target. Otherwise, the lookup
+    field will be returned verbatim.
 
-    """
-    json_api_link = True  # serializes to a links object
-
-    def __init__(self, view_name=None, **kwargs):
-        self.meta = kwargs.pop('meta', {})
-        self.link_type = kwargs.pop('link_type', 'url')
-        super(JSONAPIHyperlinkedIdentityField, self).__init__(view_name=view_name, **kwargs)
+    Example:
+     wiki_home = RelationshipField(
+        related_view='addon:addon-detail',
+        related_view_kwargs={'node_id': '<_id>', 'provider': 'wiki'},
+    )
+    '_id' is enclosed in angular brackets, but 'wiki' is not. 'id' will be looked up on the target, but 'wiki' will not.
+     The serialized result would be '/nodes/abc12/addons/wiki'.
 
-    # overrides HyperlinkedIdentityField
-    def get_url(self, obj, view_name, request, format):
-        """
-        Given an object, return the URL that hyperlinks to the object.
+    Field can handle nested attributes:
 
-        Returns None if lookup value is None
-        """
+    Example:
+    wiki_home = RelationshipField(
+        related_view='wiki:wiki-detail',
+        related_view_kwargs={'node_id': '<_id>', 'wiki_id': '<wiki_pages_current.home>'}
+    )
 
-        if getattr(obj, self.lookup_field) is None:
-            return None
+    """
 
-        return super(JSONAPIHyperlinkedIdentityField, self).get_url(obj, view_name, request, format)
+    json_api_link = True  # serializes to a links object
 
-    # overrides HyperlinkedIdentityField
-    def to_representation(self, value):
+    def __init__(self, **kwargs):
+        related_view = kwargs.pop('related_view', None)
```

Comment:
It's more readable and more concise to explicitly include argument names in the method definition rather than popping off `kwargs`. It also aids static analysis (e.g. in IDEs).

```python
    def __init__(self, related_view=None, related_view_kwargs=None, self_view=None, self_view_kwargs=None,
                 self_meta=None, related_meta=None, **kwargs):
        related_view = related_view
        self_view = self_view
        related_kwargs = related_view_kwargs
        self_kwargs = self_view_kwargs
```

I can make this change when I merge.